src/parser/parser_support.c

```diff
@@ -8523,7 +8523,11 @@ pt_check_grammar_charset_collation (PARSER_CONTEXT * parser, PT_NODE * charset_n
 	default:
 	  assert (*charset == INTL_CODESET_BINARY);
 	  *coll_id = LANG_COLL_BINARY;
-	  break;
+	  return NO_ERROR;
+	}
+      if (type == PT_TYPE_VARCHAR || type == PT_TYPE_VARNCHAR)
+	{
+	  *coll_id = *coll_id + COLL_TS;	/* for trailing space sensitive */
```

Comment:
Can the default character set be insensitive to trailing spaces?

Reply:
If the collation was specified at the syntax, this routine is not used.

Reply:
This statement is run only when the collation is not specified at query string, therefore there is no unexpected collation setting